common/src/commonMain/kotlin/io/xorum/codeforceswatcher/util/IFeedbackController.kt

```diff
@@ -0,0 +1,78 @@
+package io.xorum.codeforceswatcher.util
+
+abstract class IFeedbackController {
+    abstract var countScreenOpening: Int
+
+    abstract var startTimeWhenShown: Long
+
+    abstract var isNeededAtAll: Boolean
+
+    abstract var isLockoutPeriod: Boolean
+
+    abstract var currentShowingItem: Int
+
+    var feedbackItem: FeedbackItem
+        get() = when(currentShowingItem) {
+            0 -> buildEnjoyingItem()
+            1 -> { isNeededAtAll = false; buildEmailItem() }
+            2 -> buildRateItem()
+            else -> {
+                throw IllegalStateException()
+            }
+        }
+        set(v) {}
+
+    abstract fun buildEmailItem(): FeedbackItem
```

Comment:
You can pass `positiveButtonClick, negativeButtonClick, neutralButtonClick` as params to your abstract methods. In such way you will move all business-logic to `common`.